### ISSUE TITLE
TitleBar doesn't always hide

### ISSUE THREAD

**Issue by terencode** [velitasali/GTKTitleBar#20]
Sometimes the titlebar doesn't hide like it's supposed to do and it takes a restart of the concerned application for it to work again.

- I'm using your latest commit.

- I see this sporadically in the log:
```
déc. 01 17:33:01 terence-arch gnome-shell[1911]: X Error of failed request:  BadWindow (invalid Window parameter)
déc. 01 17:33:01 terence-arch gnome-shell[1911]:   Major opcode of failed request:  18 (X_ChangeProperty)
déc. 01 17:33:01 terence-arch gnome-shell[1911]:   Resource id in failed request:  0x2800176
déc. 01 17:33:01 terence-arch gnome-shell[1911]:   Serial number of failed request:  13
déc. 01 17:33:01 terence-arch gnome-shell[1911]:   Current serial number in output stream:  15
déc. 01 17:33:09 terence-arch gnome-shell[1911]: X Error of failed request:  BadWindow (invalid Window parameter)
déc. 01 17:33:09 terence-arch gnome-shell[1911]:   Major opcode of failed request:  18 (X_ChangeProperty)
déc. 01 17:33:09 terence-arch gnome-shell[1911]:   Resource id in failed request:  0x28001a3
déc. 01 17:33:09 terence-arch gnome-shell[1911]:   Serial number of failed request:  13
déc. 01 17:33:09 terence-arch gnome-shell[1911]:   Current serial number in output stream:  15
```

**Comment by velitasali** [comment]
Is there a specific application that you experience this with?

**Comment by terencode** [comment]
Atom and MellowPlayer.

**Comment by velitasali** [comment]
I am guessing the window does not appear in the known windows list. I will try to produce the error myself then maybe I can fix it.

**Comment by weeman1337** [comment]
For me it stopped working for PhpStorm. A previous release did work.

**Comment by velitasali** [comment]
Could they finally have brought borderless UI? If so, it is
considered to be a client decorated app.

**Comment by weeman1337** [comment]
> Could they finally have brought borderless UI?

Nope there is still the default (useless) title bar.

If I can help with any window information or logs please let me know.

**Comment by weeman1337** [comment]
I played around a bit and found out that two windows of PhpStorm behave different: one has a hidden title bar - the other hasn't.

**Comment by velitasali** [comment]
> I played around a bit and found out that two windows of PhpStorm behave different: one has a hidden title bar - the other hasn't.

How are they different from each other?

**Comment by weeman1337** [comment]
They are both main windows with a project open. ¯\\\_(ツ)\_/¯

Is there any info I could gather about these windows?

**Comment by weeman1337** [comment]
> For me it stopped working for PhpStorm. A previous release did work.

I meant a previous GTK Title Bar release. So PhpStorm didn't change. I've finally updated all my extension ;)

**Comment by velitasali** [comment]
Is it working now :+1:  ?

**Comment by weeman1337** [comment]
> Is it working now +1 ?

Nope. It doesn't. As written above if I can help with some log output or system investigation please let me know.

**Comment by velitasali** [comment]
Alright, could you answer these?
1. Which version of GNOME?
2. Is it X or Wayland session?
3. `xprop > window1.txt` and click on the first window.

4. `xprop > window2.txt` and click on the second window.
5. Upload the window1.txt and window2.txt files here.

Thanks

**Comment by weeman1337** [comment]
> 1. Which version of GNOME?

3.34.1

>  Is it X or Wayland session?

It is an X session

> Upload the window1.txt and window2.txt files here.

* [window_not_working.txt](https://github.com/velitasali/GTKTitleBar/files/3970236/window_not_working.txt)
* [window_working.txt](https://github.com/velitasali/GTKTitleBar/files/3970237/window_working.txt)

I hope that helps. If I can send any other info or test something just ask.

**Comment by MattSturgeon** [comment]
This should've been fixed by ce0a0e6f, which adds a option to disable the primary monitor check.

**Comment by weeman1337** [comment]
The limitation to the primary monitor is off. Still the title bar is there (sometimes).

Another interesting behaviour: If I lock the screen and unlock it the title bars are gone.

**Comment by velitasali** [comment]
@weeman1337 Is PhpStorm the only application you are having an issue with? I am using other JetBrains' IDEs and they don't have that kind of behavior with this extension. I would say that maybe the window you are referring to is not maximizable and thus has its title bar not hidden, but seems to be not the case since they are both different main windows.

**Comment by weeman1337** [comment]
> Is PhpStorm the only application you are having an issue with?

It happens with all the IntelliJ products in their latest version. I can also confirm that this happens under Ubuntu 19.04 and 19.10.

**Comment by velitasali** [comment]
> It happens with all the IntelliJ products in their latest version. I can also confirm that this happens under Ubuntu 19.04 and 19.10.

Hm, I am not sure. Sadly, I am not on Ubuntu. Also, do you use any proprietary graphics driver like NVIDIA's? And could you record the screen while it's happening? Ctrl + Shift + Alt + R should do it easily and save it under the Videos folder.

**Comment by terencode** [comment]
In addition to the `BadWindow (invalid Window parameter)`` error I also get 
```
Could not create transient scope for PID 1003902: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Process with ID 1003902 does not exist.
```

when hiding/showing a window.

**Comment by ph1-xyz** [comment]
Hi!

I installed the Extension downloading the latest release available (_gtktitlebar-5.0_) and moving it to the extensions' folder. I found it inside the _Tweaks Menu_, set its parameter to _Always Hide_ also on not-primary screens, rebooted, but I still seeing the Title Bars in Terminal and Inkscape (where I'd like to remove it).

I'm using GNOME Shell 3.34.5 on Fedora 31. I don't know how to check if I'm using X or Wayland. 

Thank you! Let me know if I can provide other useful details to make the extension work

**Comment by velitasali** [comment]
@ph1-xyz your issue may be similar to https://github.com/velitasali/GTKTitleBar/issues/21. You can use the same JavaScript that I used in the screenshot to check if those windows are reporting they are client decorated. 

To open the looking glass, Alt + F2, `lg`. Now run the following call and make sure the only running apps are those you have issue with. Otherwise you will get random boolean values printed.

```js
global.get_window_actors().forEach(window => window.metaWindow.is_client_decorated()) 
```
The returned values should be false. If they all true, you are having the same issue.

**Comment by ph1-xyz** [comment]
Thanks @velitasali. It seems the menu opened from Alt+f2 doesn't accept that javascript as a command

I also can't see which screenshot you're referring to

**Comment by velitasali** [comment]
@ph1-xyz no worries previous code didn't work exactly either. Run this one. I added a screencast to help you.
```js
let values = []; global.get_window_actors().forEach(window => values.push("(" + window.metaWindow.get_title() + ": " + window.metaWindow.is_client_decorated() + ")")); values
```
Note: couldn't upload webm files, so I zipped it.
[Screencast from 25-04-2020 10:40:08.zip](https://github.com/velitasali/GTKTitleBar/files/4532570/Screencast.from.25-04-2020.10.40.08.zip)

Edit: added the gif version of it
![Screencast from 25-04-2020 10_40_08](https://user-images.githubusercontent.com/10258162/80274305-174eea00-86e2-11ea-9156-e91eb6410799.gif)

**Comment by ph1-xyz** [comment]
Thanks. Didn't know about this `lg` command

They are all _true_.

**Comment by velitasali** [comment]
@ph1-xyz Great. looks like you are one of those who need the whitelist feature :smiley:

Even though, it is easy to add. It is hard to test and debug an extension on GNOME. I will work on this when I have time.

**Comment by ph1-xyz** [comment]
Ok, I don't understand much of this, but let me know if I can help

**Comment by velitasali** [comment]
@ph1-xyz you can subscribe to https://github.com/velitasali/GTKTitleBar/issues/2 to get notified when there is a change.
